src/dftbp/io/hsdparser.F90

```diff
@@ -185,6 +185,7 @@ subroutine parseHSD_opened(initRootName, fd, file, xmlDoc)
     !> DOM-tree of the parsed input on exit
     type(fnode), pointer :: xmlDoc
 
+    logical, parameter :: parsedTypes(nSeparator) = .true.
```

Comment:
Likewise, is this needed to resolve the finalizer problem, or should it be its own PR?